dynamic.rb

```diff
@@ -0,0 +1,30 @@
+def dynamicArray(n, q)
+   
+    arrys = Array.new(n.to_i){[]}
+
+#Write the code
+
+lastAnswer = 0
+
+
+q.length.times do
+
+
+     arry = ((x^ lastAns) % n)
+
+    
+    arr = arrys[arry]
+   
+    arr << y if q_type == "1"
+
+        if q_type == "2"
+       
+        idx = y % arr.length
+       
+        lastAnswer = arr[idx]
+        
+        puts lastAnswer
+    end
+end
```

Comment:
Can you please make the above code readable by having separate methods and the likes?